Source/SIMPLib/CoreFilters/CreateDataArray.cpp

```diff
@@ -429,71 +439,139 @@ void CreateDataArray::execute()
 }
 
 // -----------------------------------------------------------------------------
 //
 // -----------------------------------------------------------------------------
-template <typename T> void CreateDataArray::checkInitialization(QString dataArrayName)
+template <typename T>
+void CreateDataArray::checkInitialization(QString dataArrayName)
 {
   if(m_InitializationType == Manual)
   {
     bool ok;
-    double input = m_InitializationValue.toDouble(&ok);
-    if(!ok)
+
+    // Check to see if we have multiple values
+    bool startsWithSqBracket = m_InitializationValue.trimmed().startsWith("[");
+    bool endsWithSqBracket = m_InitializationValue.trimmed().endsWith("]");
+    // If only 1 bracket appears that is bad.
+    if((!startsWithSqBracket && endsWithSqBracket) || (startsWithSqBracket && !endsWithSqBracket))
     {
-      QString ss = "Could not convert initialization value to a double.";
-      setErrorCondition(-5559, ss);
+      QString ss = QString("Init string is not valid. Starting or ending square brackets are missing.");
+      setErrorCondition(-5458, ss);
       return;
     }
 
-    if(input < static_cast<double>(std::numeric_limits<T>().lowest()) || input > static_cast<double>(std::numeric_limits<T>().max()))
+    // If there are no brackets then the string should be able to be converted to a number.
+    // double is the easiest to sanity check because whether the value has a decimal point or not
+    // it would convert to a double. If we tried an int and the string had a decimal point
+    // the conversion would fail.
+    if(!startsWithSqBracket && !endsWithSqBracket)
     {
-      QString ss = QObject::tr("%1: Invalid initialization value. The valid range is %2 to %3").arg(dataArrayName).arg(std::numeric_limits<T>::min()).arg(std::numeric_limits<T>::max());
-      setErrorCondition(-4000, ss);
-      return;
+      QString input = m_InitializationValue.trimmed();
+      std::ignore = input.toDouble(&ok);
+      if(!ok)
+      {
+        QString ss = QString("Init string cannot be converted to a number. If this is a multicomponent surround the string with square brackets [ ]");
+        setErrorCondition(-5457, ss);
+        return;
+      }
+    }
+
+    // Just validate the number of components are valid and each token can be converted to a double
+    if(startsWithSqBracket && endsWithSqBracket)
+    {
+      QString input = m_InitializationValue.trimmed();
+      input = input.replace('[', "");
+      input = input.replace(']', "");
+      QStringList tokens = input.split(CDA_SEP, QString::SplitBehavior::KeepEmptyParts);
+      if(m_NumberOfComponents != tokens.size() && tokens.size() != 1)
+      {
+        QString ss = QString("Number of initialization components does not match number of components.%1 vs %2").arg(tokens.size()).arg(m_NumberOfComponents);
+        setErrorCondition(-5459, ss);
+        return;
+      }
+      for(const auto& token : tokens)
+      {
+        std::ignore = token.toDouble(&ok);
```

Comment:
We should check the conversion with the actual type. For example if you enter -1 for an unsigned type, it the Qt functions fail to convert it resulting in 0, but it is a silent error during dataCheck().